### DISCUSSION TITLE
Node 18 Whitespace + Self-closing Tags

### DISCUSSION THREAD

**Discussion by mcblum** [handlebars-lang/handlebars.js#1943] [Q&A] [answered]
Hello!

We're in the process of upgrading our app to use Node 18. When I did the upgrade, all of the tests failed. Looking into it, it looks like there's some difference where all of the whitespace is being collapsed and the self-closing tags stripped out. So `<meta />` is becoming `<meta>`.

I see some recent commits for testing with Node 18, but I'm wondering if it's fully supported or if anyone else has seen this?

Thanks!

**Comment by jaylinski** [comment]
Hm... the tests seem to run fine, even with Node 20:

https://github.com/handlebars-lang/handlebars.js/actions/runs/4997611619/jobs/8952151271

Also: it would be weird if the Node version somehow had influence on how Handlebars handles whitespace.

Did you upgrade `handlebars` or did it stay the same in your lock-file?

**Comment by mcblum** [comment]
@jaylinski I thought the same thing! I did not, it stayed exactly the same in the lockfile, as did all its dependencies. That was surprising to me. I did learn that actually the HTML5 spec does not require self-closing tags because it considers those elements "void", but there are a couple other strange things happening, too, like the CSS color hex codes are being transformed to all uppercase. Something is definitely weird, but I can't seem to figure out why.

**Comment by jaylinski** [comment, marked as answer]
Maybe there is another post- or pre-processing step in your setup that now changes your rendered HTML?

**Reply by mcblum** [reply]
There is/was, and I totally missed it. This can be closed, I apologize for the dumb question! It was a downstream thing that I missed.
